common/src/main/java/com/microsoft/identity/common/internal/ui/webview/challengehandlers/NonceRedirectHandler.kt

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.internal.ui.webview.challengehandlers
+
+import android.app.Activity
+import android.webkit.WebView
+import com.microsoft.identity.common.CommonRefreshTokenCredentialProvider
+import com.microsoft.identity.common.adal.internal.AuthenticationConstants
+import com.microsoft.identity.common.adal.internal.util.StringExtensions
+import com.microsoft.identity.common.logging.Logger
+import java.net.URL
+
+/**
+ * Handler for processing nonce from redirect and attaching new prt credential header on web view.
+ */
+class NonceRedirectHandler(
+    private val webview: WebView? = null,
+    private val headers: HashMap<String, String>
+) : IChallengeHandler<URL, Void> {
+    private val TAG = NonceRedirectHandler::class.java.simpleName
+
+    override fun processChallenge(input: URL) : Void? {
```

Comment:
nit: does this needs to be void?

Reply:
yes this is overriding the parent class method which needs to be Void? if null result is being returned